src/engine/Scene.ts

```diff
@@ -566,9 +566,16 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
          this.removeUIActor(entity);
          return;
       }
+      
       if (entity instanceof Actor) {
-         this._broadphase.untrack(entity.body);
-         this._removeChild(entity);
+
+         if(this._killQueue.indexOf(entity) == -1) {
```

Comment:
This might be a bigger task but it'd be great to make this a O(1) call. We have actorId, we could set `_killQueue[actorId] = true`

Reply:
I think this is too big for this issue, I'm going to recommend we add this type of change to our performance milestone as a separate issue. But O(1) kill queue lookups would be ideal.